### PR TITLE
Fix infinite loading caused when filling from another locale

### LINKED ISSUE #12002: [v4] filling from another locale
### Describe the bug

Basically when you fill from another locale the app is either crashing or infinitely loading

### Steps to reproduce the behavior

Basically you need to have multiple locales.

There are 2 cases on this bug:

Either you copy a locale on an existing entity and it creates infinite loading:
image
image

or on a not existing entity (not saved when switching locale):
image
image

On this resource i have:
- A dynamic zone with multiple components
image
- A component
image
- A relation to another field 
image
- some basic fields
image

### Expected behavior

Should be able to copy the other locale on current one correctly.

I thought this was supposed to be fixed in 4.0.2 but I had a misunderstanding.

### System

- Node.js version: 14.18.1
- NPM version: 6.14.15
- Strapi version: 4.0.0
- Database: MySQL
- Operating system: Windows 11

### PR DESCRIPTION
**Pull request by iicdii**
Signed-off-by: harimkims <harimkims@gmail.com>

### What does it do?

- I put in the necessary data that was removed after the `cleanData` execution.

### Why is it needed?

- In order to prevent crashing app.
- `Fill in from another locale` should persist some data(id, createdAt...) existed in the initial entry.

### How to test it?

1. Add `ko-KR` language in i18n settings.
2. Create an entry with `en`
3. Publish
4. Click `Korean (South Korea) (ko-KR)` in Locale
5. Input some data
6. Publish
7. Click `Fill in from another locale`
8. Select `English (en)`
9. Click `yes, fill in`

### Related issue(s)/PR(s)

Closes #12002

#### Another Issue
You can't click `save` button after executing `Fill in from another locale` even though the content changed. You need to modify some field to enable `save` button. I tried to fix this issue too but I stopped because It seems that a lot of extra work is needed.

#### Why you can't click `save` button after `Fill in from another locale`?
There are two data `initialData`, `modifiedData`. 
Dispatching `GET_DATA_SUCCEEDED` override `initialData` and `modifiedData`. However, the save button is activated only when there is a change compared to `initialData` and `modifiedData`.
I believe that changing the logic by updating `modifiedData` instead of overwriting `initialData` will solve the problem.

### CODE REVIEW

File: packages/plugins/i18n/admin/src/components/CMEditViewInjectedComponents/CMEditViewCopyLocale/index.js [hunk 1]
```diff
@@ -59,12 +59,15 @@ const Content = ({ appLocales, currentLocale, localizations, readPermissions })
...
     const requestURL = `/content-manager/collection-types/${slug}/${value}`;
 
+    setIsLoading(true);
     try {
-      setIsLoading(true);
-
       const { data: response } = await axiosInstance.get(requestURL);
 
       const cleanedData = cleanData(response, allLayoutData, localizations);
+      ['createdBy', 'updatedBy', 'publishedAt', 'id', '_id', 'createdAt'].forEach(key => {
```
Comment (soupette) [inline packages/plugins/i18n/admin/src/components/CMEditViewInjectedComponents/CMEditViewCopyLocale/index.js:67]: This field were removed on purpose as we creating a new entry they should be removed.

I don't think it is the correct fix for #12002

**Review by soupette — CHANGES_REQUESTED** [review]

File: packages/plugins/i18n/admin/src/components/CMEditViewInjectedComponents/CMEditViewCopyLocale/index.js (hunk 1 above)
Comment (iicdii) [reply packages/plugins/i18n/admin/src/components/CMEditViewInjectedComponents/CMEditViewCopyLocale/index.js:67]: That's exactly why I put `if (!initialData[key]) return;` statement in code.
When you create a new entry, it neither has nor needs those fields. But when you "edit" an existing entry you should not omit them.

Comment (derrickmehaffy) [reply packages/plugins/i18n/admin/src/components/CMEditViewInjectedComponents/CMEditViewCopyLocale/index.js:67]: `'_id'` there is no reason for this to exist anymore, it was a mongodb thing that shouldn't exist anymore

Comment (iicdii) [reply packages/plugins/i18n/admin/src/components/CMEditViewInjectedComponents/CMEditViewCopyLocale/index.js:67]: @derrickmehaffy 
Yeah, I just copied them from `cleanData`, should I remove both of them?
https://github.com/strapi/strapi/blob/48893ae3fc951b618fd8c4fdc6970e623d2c92db/packages/plugins/i18n/admin/src/components/CMEditViewInjectedComponents/CMEditViewCopyLocale/utils/cleanData.js#L16-L24

Comment (soupette) [reply packages/plugins/i18n/admin/src/components/CMEditViewInjectedComponents/CMEditViewCopyLocale/index.js:67]: @iicdii my bad :see_no_evil: you were right I misunderstood the situation the bug occurs when editing an entry...

Your code is correct we just need to remove the `_id` and then we can merge it.


Thank you for your contribution!

Comment (iicdii) [reply packages/plugins/i18n/admin/src/components/CMEditViewInjectedComponents/CMEditViewCopyLocale/index.js]: Thanks!
I just remove them and wait for passing the tests..

**Review by soupette — APPROVED** [review]

**Pull request merged**